handler.js

```diff
@@ -96,38 +96,34 @@ module.exports.createLetter = (event, _context, callback) => {
     ];
   }
 
-  request({
+  await request({
     url: process.env.SLACK_WEBHOOK_URL,
     body: slackReq,
     json: true,
     method: 'POST'
-  }, (err, _req, _res) => {
-    if (err) {
-      callback(err);
-    } else {
-      callback(null, {
-        statusCode: 200,
-        headers: {
-          'Access-Control-Allow-Origin': '*'
-        },
-        body: JSON.stringify({
-          id: submissionId
-        }),
-      });
-    }
   });
+
+  return {
+    statusCode: 200,
+    headers: {
+      'Access-Control-Allow-Origin': '*'
+    },
+    body: JSON.stringify({
+      id: submissionId
+    }),
+  };
 };
 
 module.exports.approveLetter = (event, _context, callback) => {
```

Comment:
Could you explain why you're changing `createLetter` to async but not `approveLetter`?

Reply:
There are two ways to respond to Slack. The first way is to send the desired values directly in the response to the call Slack makes to us. This is straight forward, and is how I implemented this originally. The problem is that Slack has a hard-coded timeout of 3 seconds, after which it will show an error in the channel. So if we're calling out to a bunch of services (NationBuilder, SES, database, etc), this can sometimes happen.

The second way is to send a quick 200 response to the call that Slack makes, then do whatever we need to do in the background, and send another request back to Slack with the response.

In an `async` function, the response is returned at the same time as the function terminates. With regular, callback-based functions, the response can be returned (by calling the callback), and then other work can be done after.

The call to `createLetter` is done from the browser which can wait longer than 3s. So I left it to send a synchronous response (which, confusingly, means that it can use `async`). The call to `approveLetter` is done by Slack, so it's returning the response first, then doing most of the work. This means that the `callback` has to be called before the function terminates.